data_validation/__main__.py

```diff
@@ -314,18 +312,24 @@ def build_config_from_args(args: Namespace, config_manager: ConfigManager):
             _get_calculated_config(args, config_manager)
         )
 
-        # Append Comparison fields
-        if args.comparison_fields:
-            config_manager.append_comparison_fields(
-                _get_comparison_config(args, config_manager)
-            )
-
         # Append primary_keys
-        primary_keys = cli_tools.get_arg_list(args.primary_keys)
+        primary_keys = (
+            cli_tools.get_arg_list(args.primary_keys)
+            or config_manager.auto_list_primary_keys()
+        )
+        if not primary_keys:
+            raise ValueError("--primary-keys argument is required for this validation")
```

Comment:
Let us be more explicit here. Perhaps something like "No primary keys were provided on the command line and neither the source or target tables have primary keys. Please specify primary keys using --primary-keys argument"